src/components/Table/useSortState.ts

```diff
@@ -33,21 +33,42 @@ export function useSortState<R extends Kinded, S>(
     } else {
       return sorting?.value;
     }
-  });
+  }, [sorting, columns]);
+  const [sortState, setSortState] = useState<SortState<S> | undefined>(initialSortValue);
 
-  // Make a custom setSortKey that is useState-like but contains the invert-if-same-column-clicked-twice logic.
+  // Make a custom setSortKey that is useState-like but contains the ASC->DESC->RESET logic.
   const setSortKey = useCallback(
     (clickedKey: S) => {
       const [currentKey, currentDirection] = sortState || [];
-      const [newKey, newDirection] =
-        // If clickedKey === currentKey, then toggle direction
-        clickedKey === currentKey
-          ? [currentKey, currentDirection === ASC ? DESC : ASC]
-          : // Otherwise, use the new key, and default to ascending
-            [clickedKey, ASC];
-      setSortState([newKey, newDirection]);
-      if (sorting?.on === "server") {
-        sorting.onSort(newKey, newDirection);
+
+      // If sorting on a new key -> ASC this key.
+      if (clickedKey !== currentKey) {
+        setSortState([clickedKey, ASC]);
+
+        if (sorting?.on === "server") {
+          sorting.onSort(clickedKey, ASC);
+        }
+        return;
+      }
```

Comment:
Could we have a helper method like:

```
const [newKey, newDirection] = deriveSortState(currentKey, currentDirection clickedKey);
setSortState([newKey, newDirection])
if (sorting?.on === "server') {
  sorting.onSort(newKey, newDirection);
```

I.e. then we could have one `sorting?.on` + `sorting.onSort` call, which I think each of these three conditions is currently doing on its own.